iroh-net/src/magicsock.rs

```diff
@@ -563,26 +546,30 @@ impl MagicSock {
                     .unwrap_or_default();
                 if udp_pending && relay_pending {
                     // Handle backpressure.
-                    return Err(io::Error::new(io::ErrorKind::WouldBlock, "pending"));
-                }
-
-                if !relay_sent && !udp_sent && !pings_sent {
-                    // Returning Ok here means we let QUIC timeout, same with returning any
-                    // errors.  The philosophy of quinn-udp is that a UDP connection could
-                    // come back at any time so these errors should be treated as transient
-                    // and are just timeouts.  Hence we opt for returning Ok.  See
-                    // test_try_send_no_udp_addr_or_relay_url to explore this further.
-                    error!("no UDP or relay paths available for node");
-                    return Ok(());
+                    Err(io::Error::new(io::ErrorKind::WouldBlock, "pending"))
+                } else {
+                    if relay_sent || udp_sent {
+                        trace!(
+                            node = %node_id.fmt_short(),
+                            send_udp = ?udp_addr,
+                            send_relay = ?relay_url,
+                            "sent transmit",
+                        );
+                    }
+                    if !relay_sent && !udp_sent && !pings_sent {
```

Comment:
```suggestion
                    } else if !pings_sent {
```
Reads a little clearer to me